common/flatpak-dir.c

```diff
@@ -7718,6 +7814,72 @@ flatpak_dir_create_system_child_repo (FlatpakDir   *self,
   return flatpak_dir_create_child_repo (self, cache_dir, file_lock, optional_commit, error);
 }
 
+static gboolean
+flatpak_dir_setup_revokefs_fuse_mount (FlatpakDir    *self,
+                                       const gchar   *ref,
+                                       const gchar   *opt_commit,
+                                       const gchar   *installation,
+                                       gchar        **src_dir,
+                                       gchar        **mnt_dir,
+                                       GCancellable  *cancellable)
+{
+  g_autoptr (GError) local_error = NULL;
+  g_autofree gchar *src_dir_tmp = NULL;
+  g_autofree gchar *mnt_dir_tmp = NULL;
+  gint socket = -1;
+
+  if (!flatpak_dir_system_helper_call_get_revokefs_fd (self,
+                                                       FLATPAK_HELPER_GET_REVOKEFS_FD_FLAGS_NONE,
+                                                       ref,
+                                                       opt_commit,
+                                                       installation ? installation : "",
+                                                       &socket,
+                                                       &src_dir_tmp,
+                                                       cancellable,
+                                                       &local_error))
+    {
+      if (g_error_matches (local_error, G_DBUS_ERROR, G_DBUS_ERROR_NOT_SUPPORTED))
+        g_warning ("revokefs-fuse not supported on your installation: %s", local_error->message);
+      else
+        g_warning ("Failed to get revokefs-fuse socket from system-helper: %s", local_error->message);
+
+      return FALSE;
+    }
+  else
+    {
+      g_autoptr(GSubprocess) revokefs_fuse = NULL;
+      g_autoptr(GSubprocessLauncher) launcher = NULL;
+      g_autofree gchar *client_uid = NULL;
+
+      *src_dir = g_strdup (src_dir_tmp);
```

Comment:
Only set this at the end, or we will leak it on error. Also, better to us g_steal_pointer() to avoid duplicating it.